src/python/pants/testutil/engine/util.py

```diff
@@ -228,5 +239,139 @@ def fmt_rule(
       f"Get[{product_subject_pair[0]}]({product_subject_pair[1]})"
       for product_subject_pair in gets
     )
-    gets_str = f", gets=[{get_members}]"
+    gets_str = f",\ngets=[{get_members}]"
   return f"@rule({fmt_rust_function(rule)}({params}) -> {product}{gets_str})"
+
+
+def fmt_graph_rule(rule: Callable, *, gets: Optional[List[Tuple[str, str]]] = None) -> str:
```

Comment:
Very helpful utils! I think that we want these new ones to live in `test_rules.py`, though, as they don't seem generalizable to any other files. Specifically, I don't think we expect to check the output of the rule graph visualization anywhere else.

`testutil/engine/util` is a public API so moving this to `test_rules.py` will allow us a little more flexibility in changing this API, keeps this util file a bit smaller, and moves the logic closer to where it's used (aka proximity).

Reply:
Totally agree!